### PR TITLE
Update Wiimmfi patch

### PR DESCRIPTION
**Pull request by Leseratte10**
Two days ago, we've again updated all the Wiimmfi patchers to fix a security issue in a bunch of Wii games. 

This PR includes said security bugfix for non-MKWii games, and the error 5142x bugfix for Mario Kart Wii.

I'm marking this as a draft PR because it's been a long time ago that I last verified that this is working correctly, and I would like to do some more tests before this is merged and officially released; however I'm already making this PR in case other people would like to test, too.

### CODE REVIEW

**Comment by urmum-69** [comment]
The ci here seemed to fail, and I'm too lazy to set up an older devkitPPC version to compile it
Can you provide a build for it?

**Comment by wiidev** [comment]
My action permissions were limiting things so this didn't build. I've adjusted them now though.

Maybe change `char gt2locator[] = { 0x38, 0x61, 0x00, 0x08, 0x38, 0xA0, 0x00, 0x14};` to `char gt2locator[] = { 0x38, 0x61, 0x00, 0x08, 0x38, 0xA0, 0x00, 0x14 };` and then force push?

**Comment by Leseratte10** [comment]
Force-pushed the PR again. Github CI build seems to be running now, also I added a Dockerfile to the repo that allows everyone to build a local copy without Github's help, you just need Docker on your machine.

**Comment by Leseratte10** [comment]
Ci was successful - you can now click on the "Checks" tab on top of the PR, then click on "Artifacts" and download the pre-built version. 

Or run `DOCKER_BUILDKIT=1 docker build -o output .` to use Docker to compile the loader.

**Comment by urmum-69** [comment]
Following up from https://github.com/Fledge68/WiiFlow_Lite/pull/247
Another person tried both animal crossing and tetris party live on both loaders, and didn't have problems
It is probably fine to merge now, but I think a wider range of games tested would still be a good idea

File: source/patches/gamepatches.c [hunk 1]
```diff
@@ -243,6 +250,231 @@ void PrivateServerPatcher(void *addr, u32 len, u8 privateServer, const char *ser
...
+		// and applies all the necessary security fixes to the game. 
+
+		// This function has been implemented by Leseratte. Please don't
+		// try to modify it without speaking to the Wiimmfi team because
+		// doing so could have unintended side effects. 
+
+		// This function MUST not run for MKWii, that would break stuff. 
+
+		int hasGT2Error = 0;
+        char gt2locator[] = { 0x38, 0x61, 0x00, 0x08, 0x38, 0xA0, 0x00, 0x14};
```
Comment (eku) [inline source/patches/gamepatches.c:289]: I would like to note that the indentation is not consistent (space versus tab).

Comment (Leseratte10) [reply source/patches/gamepatches.c:289]: Oops. I'll update that later today.

Comment (wiidev) [reply source/patches/gamepatches.c]: I had a little free time, so I formatted the whole file.

**Comment by Leseratte10** [comment]
Welp, good thing this hasn't been merged yet. We've found another issue and had to update all the existing patchers again. Will be updating this PR asap.

**Comment by eku** [comment]
What is the current status? When will there be a version of the USB Loader that includes the patches and you can go online with the Wii games again without getting an error code?

**Comment by wiidev** [comment]
I'm still dealing with some health issues, but I believe they're still working on this anyway.

Once the PR is ready I'll check it over, merge it, push some other changes and then I'll have a new release ready soon after.

**Comment by Leseratte10** [comment]
Correct, still working on it, there will be another Wiimmfi patcher update to fix a bug in some games. Hoping that that'll be ready within the next couple days, and then I'll add the fix to this PR, too.

**Comment by Leseratte10** [comment]
Okay, with this latest commit the PR is now up-to-date and has the same Wiimmfi patch code as the most recent version of the official Wiimmfi patchers. Given that these just have been released today with another bugfix, I'd suggest waiting a couple more days before finally merging this, just in case more issues come up.

**Comment by phimble** [comment]
Will there be a build for this eventually or is this going to be locked to dockerfile? Just wondering as I'm having problems running docker.

**Comment by Leseratte10** [comment]
Okay, I would say that this should now finally be ready to be merged. There haven't been any complaints since the last update, so it's probably unlikely that there are still more hidden bugs in the patcher(s). Of course that doesn't necessarily mean there also are no bugs in this USB Loader version, but the code is almost 1:1 been taken from the other Wiimmfi patcher so I'd say that that's unlikely, too.

**Comment by laguill** [comment]
Hi,
It is the first time I try wiimfi. 
I patched COD Worldat war iso game using wiimmfi -patcher-v7.5. 
I get this error when I want to play online "Error 23500"
Is it an error of wiimfi patcher update or is it something else ?

**Comment by Leseratte10** [comment]
Okay, if you used the Wiimmfi ISO patcher that does not really have anything to do with this PR. 
CoD: World At War uses a server operated by InvoxiPlayGames, not the "normal" Wiimmfi server operated by Wiimm and me, so maybe something's currently broken with that server.

**Comment by phimble** [comment]
Ok so mostly from what I've tried it works fine, however when I try to play online with Sonic All Stars Racing it comes up with error code 60000 Communication error. You have been disconnected from Nintendo Wi-Fi Connection. Please try again. This is so far the only game I have had an issue with.

**Comment by Leseratte10** [comment]
Error 60000 means you're trying to use a friend code that Wiimmfi doesn't know; you'd need to either use a new savegame or use a cheat code once to get rid of this error.

**Pull request merged**